### DISCUSSION TITLE
The output came out as a Tex font metrics file

### DISCUSSION THREAD

**Discussion by smthjhn** [streamlink/streamlink#4231] [Q&A] [answered]
### Checklist

- [X] This is a bug report and not a different kind of issue
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed bug reports](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22bug%22)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink/commits/master)

### Streamlink version

Latest stable release

### Description

Using the latest release from pip `streamlink 3.0.3`, I used it to download a stream from twitch. They were starting and stopping the stream multiple times. The first output came out as a proper video file and I can play it but for the succeeding downloads the output came out as TeX font metrics file according to Thunar. The last message I saw before the download ended is `error: Error when reading from stream: Read timeout, exiting`. The file sizes of the broken files are around 31.4MB, 66.8MB, and 4.7MB and I tried watching them with mpv but it just says that it failed to open but with VLC it plays but there is no audio and the video is just a bunch of mostly(98%) solid colors that doesn't resemble what was streamed at all. Is there any way I can repair the file?

The command I used is `while [ 1 ]; do streamlink  --retry-streams 10 -o $(date +%s) https://www.twitch.tv/{id} best; done`

### Debug log

```text
Unfortunately I wasn't using --loglevel or any kinds of logging when I ran the script, this is all I can get from my latest output:

tvnw.net/v1/playlist/CqMEHhTf_5XTo2bvClSUc7JrasUM0bhuBOjc0K5bFBBzUndhRHfuxYCZAqxlBebekli5u4OmaG-RcDzDMOC11qVBqnxfXduTex3DMpAJontx6DfA695AC0N0QCAxn_A3ZrOsZuuYNqulySXfdhk6ymYOUzuBi9red-tLlx8zGS5b9ellEqxXy1APLvEk6iEhD3SFGit5wzzrQcZsI9gbBzQPOtrYZmznlU86xTPE-skvV7w4nwWdZ4_yF53r9bhDP9fD4E_xQQr9nCfeK7xYbjz_RhiFNj5LK6wAHe2goggQ5hr9q7N1G14NxLX3CwJQC3YzSFlh4DAr0iiN7bj6TsROFJmT7_noRrECt4-b1uTMH1zFOUXi_wsrgbiOkQ2jtXe-i_C_GaWtO5npthgjgniY7IEO9FwK1i0oKamXrdvXr5XpJmf1sE9CFe8aZGAAR_dw6TrNgp9KmzmRA07Ecz7Xb7JJkmTzeava_N2itgiKbom_psG3NkSeVAQvPvaVxanNvjPtfcpP4Ltvx8Gr2B0dFhMcsxzd46AlVa9YIbL6nZNGPxF6RvpTrTHMlgVgahFE3KGn0AuxuseUhX3gga0w8TP6v5sdJiTZ-zZleYzLR63gwanpOa73Jh_T99o9PvPKmgxOBD89VOkL0Ir9vVpNF01vQIzeeFCSoBAuefsBjunA9CwSSbHfSIvJWA0h-e-kp04maR6npsQnUH0-YCaGkzXpmBoMcCkG25bCpOh6D_nqIAEqCXVzLWVhc3QtMjC3Ag.m3u8 (404 Client Error: Not Found for url: https://video-weaver.lax03.hls.ttvnw.net/v1/playlist/CqMEHhTf_5XTo2bvClSUc7JrasUM0bhuBOjc0K5bFBBzUndhRHfuxYCZAqxlBebekli5u4OmaG-RcDzDMOC11qVBqnxfXduTex3DMpAJontx6DfA695AC0N0QCAxn_A3ZrOsZuuYNqulySXfdhk6ymYOUzuBi9red-tLlx8zGS5b9ellEqxXy1APLvEk6iEhD3SFGit5wzzrQcZsI9gbBzQPOtrYZmznlU86xTPE-skvV7w4nwWdZ4_yF53r9bhDP9fD4E_xQQr9nCfeK7xYbjz_RhiFNj5LK6wAHe2goggQ5hr9q7N1G14NxLX3CwJQC3YzSFlh4DAr0iiN7bj6TsROFJmT7_noRrECt4-b1uTMH1zFOUXi_wsrgbiOkQ2jtXe-i_C_GaWtO5npthgjgniY7IEO9FwK1i0oKamXrdvXr5XpJmf1sE9CFe8aZGAAR_dw6TrNgp9KmzmRA07Ecz7Xb7JJkmTzeava_N2itgiKbom_psG3NkSeVAQvPvaVxanNvjPtfcpP4Ltvx8Gr2B0dFhMcsxzd46AlVa9YIbL6nZNGPxF6RvpTrTHMlgVgahFE3KGn0AuxuseUhX3gga0w8TP6v5sdJiTZ-zZleYzLR63gwanpOa73Jh_T99o9PvPKmgxOBD89VOkL0Ir9vVpNF01vQIzeeFCSoBAuefsBjunA9CwSSbHfSIvJWA0h-e-kp04maR6npsQnUH0-YCaGkzXpmBoMcCkG25bCpOh6D_nqIAEqCXVzLWVhc3QtMjC3Ag.m3u8)
[stream.hls][warning] Failed to reload playlist: Unable to open URL: https://video-weaver.lax03.hls.ttvnw.net/v1/playlist/CqMEHhTf_5XTo2bvClSUc7JrasUM0bhuBOjc0K5bFBBzUndhRHfuxYCZAqxlBebekli5u4OmaG-RcDzDMOC11qVBqnxfXduTex3DMpAJontx6DfA695AC0N0QCAxn_A3ZrOsZuuYNqulySXfdhk6ymYOUzuBi9red-tLlx8zGS5b9ellEqxXy1APLvEk6iEhD3SFGit5wzzrQcZsI9gbBzQPOtrYZmznlU86xTPE-skvV7w4nwWdZ4_yF53r9bhDP9fD4E_xQQr9nCfeK7xYbjz_RhiFNj5LK6wAHe2goggQ5hr9q7N1G14NxLX3CwJQC3YzSFlh4DAr0iiN7bj6TsROFJmT7_noRrECt4-b1uTMH1zFOUXi_wsrgbiOkQ2jtXe-i_C_GaWtO5npthgjgniY7IEO9FwK1i0oKamXrdvXr5XpJmf1sE9CFe8aZGAAR_dw6TrNgp9KmzmRA07Ecz7Xb7JJkmTzeava_N2itgiKbom_psG3NkSeVAQvPvaVxanNvjPtfcpP4Ltvx8Gr2B0dFhMcsxzd46AlVa9YIbL6nZNGPxF6RvpTrTHMlgVgahFE3KGn0AuxuseUhX3gga0w8TP6v5sdJiTZ-zZleYzLR63gwanpOa73Jh_T99o9PvPKmgxOBD89VOkL0Ir9vVpNF01vQIzeeFCSoBAuefsBjunA9CwSSbHfSIvJWA0h-e-kp04maR6npsQnUH0-YCaGkzXpmBoMcCkG25bCpOh6D_nqIAEqCXVzLWVhc3QtMjC3Ag.m3u8 (404 Client Error: Not Found for url: https://video-weaver.lax03.hls.ttvnw.net/v1/playlist/CqMEHhTf_5XTo2bvClSUc7JrasUM0bhuBOjc0K5bFBBzUndhRHfuxYCZAqxlBebekli5u4OmaG-RcDzDMOC11qVBqnxfXduTex3DMpAJontx6DfA695AC0N0QCAxn_A3ZrOsZuuYNqulySXfdhk6ymYOUzuBi9red-tLlx8zGS5b9ellEqxXy1APLvEk6iEhD3SFGit5wzzrQcZsI9gbBzQPOtrYZmznlU86xTPE-skvV7w4nwWdZ4_yF53r9bhDP9fD4E_xQQr9nCfeK7xYbjz_RhiFNj5LK6wAHe2goggQ5hr9q7N1G14NxLX3CwJQC3YzSFlh4DAr0iiN7bj6TsROFJmT7_noRrECt4-b1uTMH1zFOUXi_wsrgbiOkQ2jtXe-i_C_GaWtO5npthgjgniY7IEO9FwK1i0oKamXrdvXr5XpJmf1sE9CFe8aZGAAR_dw6TrNgp9KmzmRA07Ecz7Xb7JJkmTzeava_N2itgiKbom_psG3NkSeVAQvPvaVxanNvjPtfcpP4Ltvx8Gr2B0dFhMcsxzd46AlVa9YIbL6nZNGPxF6RvpTrTHMlgVgahFE3KGn0AuxuseUhX3gga0w8TP6v5sdJiTZ-zZleYzLR63gwanpOa73Jh_T99o9PvPKmgxOBD89VOkL0Ir9vVpNF01vQIzeeFCSoBAuefsBjunA9CwSSbHfSIvJWA0h-e-kp04maR6npsQnUH0-YCaGkzXpmBoMcCkG25bCpOh6D_nqIAEqCXVzLWVhc3QtMjC3Ag.m3u8)
error: Error when reading from stream: Read timeout, exiting
```

**Comment by bastimeyer** [comment, marked as answer]
> but for the succeeding downloads the output came out as TeX font metrics file according to Thunar

This doesn't mean anything, because your file manager is simply trying to guess the file format which it can't detect properly.

Whatever happend to your file output, the file container is probably corrupted. Try to fix this with FFmpeg, eg. by simply remuxing the file.

```
$ ffmpeg -i input.ts -c copy output.mp4
```

HLS streams are simple MPEG-TS streams which consist of many chunks/segments that get concatenated by Streamlink. This is the same for the player output and the file output. If a stream stops regularly or when a read timeout occurs, then the output simply stops, but that isn't a problem with the way HLS streams work.
